src/main/java/de/rwth/idsg/steve/repository/WebUserRepository.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SteVe - SteckdosenVerwaltung - https://github.com/steve-community/steve
+ * Copyright (C) 2013-2024 SteVe Community Team
+ * All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package de.rwth.idsg.steve.repository;
+
+import org.springframework.security.provisioning.UserDetailsManager;
+
+public interface WebUserRepository extends UserDetailsManager {
```

Comment:
That's a bit weird to hide a high-level API like UserDetailsManager behind a low-level API like a repository.

I mean, from a conceptual point of view, it sounds logical that a UserDetailsManager uses a repository but it sounds weird that a repository "is a" UserDetailsManager.
It is like `JdbcUserDetailsManager` extending `JdbcTemplate` instead of using it.

WDYT?

Reply:
i agree with you.

at first i was also confused when i looked at the interfaces UserDetailsService and UserDetailsManager, and their implementations, such as JdbcDaoImpl and JdbcUserDetailsManager. JdbcDaoImpl and JdbcUserDetailsManager are pure repository implementations.

then, i implemented the necessary functions in WebUserRepositoryImpl and realized that like 95% of the code is DB access only. then i stopped worrying.

i can add some additional ceremony by adding WebUserService that uses a WebUserRepository etc. but a  WebUserService would just forward the call in most cases.

Reply:
I think you should. At least it will be straight forward when you read the implementation.

Reply:
done in https://github.com/steve-community/steve/pull/1539/commits/0992d5cdc3f13ead84e1d09157b8d1bfdd80bfde